### ISSUE TITLE
yarn build failed on netlify

### ISSUE THREAD

**Issue by thdoan** [chec/commercejs-nextjs-demo-store#141]
After verifying that the demo store functions fine on http://localhost:3000, I followed the instructions in the README to deploy to Netlify, but it failed. Here is the log:

```
5:07:16 PM: Build ready to start
5:07:18 PM: build-image version: b0258b965567defc4a2d7e2f2dec2e00c8f73ad6
5:07:18 PM: build-image tag: v3.4.1
5:07:18 PM: buildbot version: 19f1164460e8d8241255cf2a71abc5e0e454118e
5:07:18 PM: Fetching cached dependencies
5:07:18 PM: Failed to fetch cache, continuing with build
5:07:18 PM: Starting to prepare the repo for build
5:07:19 PM: No cached dependencies found. Cloning fresh repo
5:07:19 PM: git clone https://github.com/ocmyusa/store
5:07:20 PM: Preparing Git Reference refs/heads/main
5:07:21 PM: Starting build script
5:07:21 PM: Installing dependencies
5:07:21 PM: Python version set to 2.7
5:07:23 PM: v12.18.0 is already installed.
5:07:23 PM: Now using node v12.18.0 (npm v6.14.4)
5:07:23 PM: Started restoring cached build plugins
5:07:23 PM: Finished restoring cached build plugins
5:07:24 PM: Attempting ruby version 2.7.1, read from environment
5:07:25 PM: Using ruby version 2.7.1
5:07:26 PM: Using PHP version 5.6
5:07:26 PM: 5.2 is already installed.
5:07:26 PM: Using Swift version 5.2
5:07:26 PM: Started restoring cached node modules
5:07:26 PM: Finished restoring cached node modules
5:07:26 PM: Installing NPM modules using NPM version 6.14.4
5:07:30 PM: npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues/3142
5:07:34 PM: npm WARN deprecated har-validator@5.1.5: this library is no longer supported
5:07:38 PM: npm WARN deprecated core-js@2.6.11: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
5:07:39 PM: npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
5:07:41 PM: npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
5:07:41 PM: npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
5:07:42 PM: npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
5:08:08 PM: > node-sass@4.14.1 install /opt/build/repo/node_modules/node-sass
5:08:08 PM: > node scripts/install.js
5:08:08 PM: Downloading binary from https://github.com/sass/node-sass/releases/download/v4.14.1/linux-x64-72_binding.node
5:08:09 PM: Download complete
5:08:09 PM: Binary saved to /opt/build/repo/node_modules/node-sass/vendor/linux-x64-72/binding.node
5:08:09 PM: Caching binary to /opt/buildhome/.npm/node-sass/4.14.1/linux-x64-72_binding.node
5:08:09 PM: > core-js@2.6.11 postinstall /opt/build/repo/node_modules/core-js
5:08:09 PM: > node -e "try{require('./postinstall')}catch(e){}"
5:08:09 PM: > core-js-pure@3.6.5 postinstall /opt/build/repo/node_modules/core-js-pure
5:08:09 PM: > node -e "try{require('./postinstall')}catch(e){}"
5:08:10 PM: > node-sass@4.14.1 postinstall /opt/build/repo/node_modules/node-sass
5:08:10 PM: > node scripts/build.js
5:08:10 PM: Binary found at /opt/build/repo/node_modules/node-sass/vendor/linux-x64-72/binding.node
5:08:10 PM: Testing binary
5:08:10 PM: Binary is fine
5:08:10 PM: > swiper@5.4.5 postinstall /opt/build/repo/node_modules/swiper
5:08:10 PM: > echo "Love Swiper? Support Vladimir's work by donating or pledging on patreon: > https://patreon.com/vladimirkharlampidi
5:08:10 PM: "
5:08:10 PM: Love Swiper? Support Vladimir's work by donating or pledging on patreon: > https://patreon.com/vladimirkharlampidi
5:08:12 PM: npm notice created a lockfile as package-lock.json. You should commit this file.
5:08:12 PM: npm WARN notsup Unsupported engine for react-lottie@1.2.3: wanted: {"npm":"^3.0.0"} (current: {"node":"12.18.0","npm":"6.14.4"})
5:08:12 PM: npm WARN notsup Not compatible with your version of node/npm: react-lottie@1.2.3
5:08:12 PM: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@~2.1.2 (node_modules/chokidar/node_modules/fsevents):
5:08:12 PM: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
5:08:12 PM: npm WARN notsup Unsupported engine for watchpack-chokidar2@2.0.0: wanted: {"node":"<8.10.0"} (current: {"node":"12.18.0","npm":"6.14.4"})
5:08:12 PM: npm WARN notsup Not compatible with your version of node/npm: watchpack-chokidar2@2.0.0
5:08:12 PM: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@^1.2.7 (node_modules/watchpack-chokidar2/node_modules/chokidar/node_modules/fsevents):
5:08:12 PM: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
5:08:12 PM: npm WARN tsutils@3.17.1 requires a peer of typescript@>=2.8.0 || >= 3.2.0-dev || >= 3.3.0-dev || >= 3.4.0-dev || >= 3.5.0-dev || >= 3.6.0-dev || >= 3.6.0-beta || >= 3.7.0-dev || >= 3.7.0-beta but none is installed. You must install peer dependencies yourself.
5:08:12 PM: npm WARN commercejs-nextjs-demo-store@0.1.0 No repository field.
5:08:12 PM: added 1533 packages from 536 contributors and audited 1536 packages in 44.814s
5:08:13 PM: 63 packages are looking for funding
5:08:13 PM:   run `npm fund` for details
5:08:13 PM: found 4 vulnerabilities (2 low, 2 high)
5:08:13 PM:   run `npm audit fix` to fix them, or `npm audit` for details
5:08:13 PM: NPM modules installed
5:08:13 PM: Started restoring cached go cache
5:08:13 PM: Finished restoring cached go cache
5:08:13 PM: go version go1.14.4 linux/amd64
5:08:13 PM: go version go1.14.4 linux/amd64
5:08:13 PM: Installing missing commands
5:08:13 PM: Verify run directory
5:08:15 PM: ​
5:08:15 PM: ┌─────────────────────────────┐
5:08:15 PM: │        Netlify Build        │
5:08:15 PM: └─────────────────────────────┘
5:08:15 PM: ​
5:08:15 PM: ❯ Version
5:08:15 PM:   @netlify/build 5.1.0
5:08:15 PM: ​
5:08:15 PM: ❯ Flags
5:08:15 PM:   deployId: 5f9caab436a6754286d1116f
5:08:15 PM:   mode: buildbot
5:08:15 PM: ​
5:08:15 PM: ❯ Current directory
5:08:15 PM:   /opt/build/repo
5:08:15 PM: ​
5:08:15 PM: ❯ Config file
5:08:15 PM:   /opt/build/repo/netlify.toml
5:08:15 PM: ​
5:08:15 PM: ❯ Context
5:08:15 PM:   production
5:08:15 PM: ​
5:08:15 PM: ❯ Loading plugins
5:08:15 PM:    - netlify-plugin-onboarding@1.0.0 from netlify.toml and package.json
5:08:15 PM: ​
5:08:15 PM: ┌──────────────────────────────────────────────────────┐
5:08:15 PM: │ 1. onPreBuild command from netlify-plugin-onboarding │
5:08:15 PM: └──────────────────────────────────────────────────────┘
5:08:15 PM: ​
5:08:15 PM: Cloning into '/opt/build/repo/node_modules/netlify-plugin-onboarding/template'...
5:08:26 PM: > core-js@2.6.11 postinstall /opt/build/repo/node_modules/netlify-plugin-onboarding/template/node_modules/core-js
5:08:26 PM: > node -e "try{require('./postinstall')}catch(e){}"
5:08:27 PM: > ejs@2.7.4 postinstall /opt/build/repo/node_modules/netlify-plugin-onboarding/template/node_modules/ejs
5:08:27 PM: > node ./postinstall.js
5:08:27 PM: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.3 (node_modules/chokidar/node_modules/fsevents):
5:08:27 PM: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.3: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
5:08:27 PM: npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.13 (node_modules/fsevents):
5:08:27 PM: npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.13: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
5:08:27 PM: added 528 packages from 405 contributors and audited 533 packages in 10.866s
5:08:27 PM: 6 packages are looking for funding
5:08:27 PM:   run `npm fund` for details
5:08:27 PM: found 9 vulnerabilities (7 low, 2 high)
5:08:27 PM:   run `npm audit fix` to fix them, or `npm audit` for details
5:08:28 PM: > example-onboarding-flow@1.0.0 build /opt/build/repo/node_modules/netlify-plugin-onboarding/template
5:08:28 PM: > eleventy --input src --output dist
5:08:28 PM: Writing dist/index.html from ./src/index.md.
5:08:28 PM: Wrote 1 file in 0.11 seconds (v0.11.0)
5:08:28 PM: ​
5:08:28 PM: (netlify-plugin-onboarding onPreBuild completed in 13.1s)
5:08:28 PM: ​
5:08:28 PM: ┌────────────────────────────────────┐
5:08:28 PM: │ 2. build.command from netlify.toml │
5:08:28 PM: └────────────────────────────────────┘
5:08:28 PM: ​
5:08:28 PM: $ yarn deploy
5:08:28 PM: bash: yarn: command not found
5:08:28 PM: ​
5:08:28 PM: ┌─────────────────────────────┐
5:08:28 PM: │   "build.command" failed    │
5:08:28 PM: └─────────────────────────────┘
5:08:28 PM: ​
5:08:28 PM:   Error message
5:08:28 PM:   Command failed with exit code 127: yarn deploy
5:08:28 PM: ​
5:08:28 PM:   Error location
5:08:28 PM:   In build.command from netlify.toml:
5:08:28 PM:   yarn deploy
5:08:28 PM: ​
5:08:28 PM:   Resolved config
5:08:28 PM:   build:
5:08:28 PM:     command: yarn deploy
5:08:28 PM:     commandOrigin: config
5:08:28 PM:     publish: /opt/build/repo/out
5:08:28 PM:   plugins:
5:08:28 PM:     - inputs: {}
5:08:28 PM:       origin: config
5:08:28 PM:       package: netlify-plugin-onboarding
5:08:28 PM: Caching artifacts
5:08:28 PM: Started saving node modules
5:08:28 PM: Finished saving node modules
5:08:28 PM: Started saving build plugins
5:08:28 PM: Finished saving build plugins
5:08:28 PM: Started saving pip cache
5:08:29 PM: Finished saving pip cache
5:08:29 PM: Started saving emacs cask dependencies
5:08:29 PM: Finished saving emacs cask dependencies
5:08:29 PM: Started saving maven dependencies
5:08:29 PM: Finished saving maven dependencies
5:08:29 PM: Started saving boot dependencies
5:08:29 PM: Finished saving boot dependencies
5:08:29 PM: Started saving go dependencies
5:08:29 PM: Finished saving go dependencies
5:08:31 PM: Build failed due to a user error: Build script returned non-zero exit code: 2
5:08:31 PM: Failing build: Failed to build site
5:08:31 PM: Failed during stage 'building site': Build script returned non-zero exit code: 2
5:08:31 PM: Finished processing build request in 1m13.431100281s
```

Could it be because the GitHub repo is private?

**Comment by thdoan** [comment]
UPDATE: I added the CHEC_PUBLIC_KEY environment variable and re-deployed, but same 127 exit code.

**Comment by thdoan** [comment]
UPDATE 2: I identified the root cause -- it was because I followed the recommended instructions on the Yarn website and installed Yarn from the msi file since I'm on Windows. Since I never ran `npm install yarn`, it was never added to `package.json`, so when Netlify runs `yarn deploy` it returned the "yarn: command not found" error. After pushing an updated `package.json` containing `"yarn": "^1.22.10"` (which was added automatically when I ran `npm install yarn`), the deployment via Netlify ran without a hitch :-).
